### ISSUE TITLE
Preserve realigned bam file when using gatk for genotyping

### ISSUE THREAD

**Issue by aidaanva** [nf-core/eager#444]
For both visual inspection of bam files and upstream analysis, it will be great if the realigned bam file can be kept when running the gatk genotyping.

**Comment by jfy133** [comment]
To stop cluttering up output, I will make this optional with a flag, so a user can explicitly call it.

**Comment by jfy133** [comment]
Same context - add check that a JAR is actually supplied (ending in jar)

**Comment by aidaanva** [comment]
Thank you for adding the flag. I have an extra request, would it be possible that when the flag is specified the bam file gets also indexed? We usually use the realigned bam files for visual inspection in IGV and that would be useful.

**Comment by apeltzer** [comment]
Hi @aidaanva ! Yes, sure. Should be easy to have something like this:

```
def run_indexing = params.saveIntermediates ? 'samtoolx index bamfile' : ''
script
"""
...
$run_indexing
"""
```

(and have an optional output statement in the `output:` section of that particular step 👍

**Comment by apeltzer** [comment]
Should be fixed in #495
